ambari/src/main/java/io/brooklyn/ambari/AmbariClusterImpl.java

```diff
@@ -79,47 +59,52 @@
     public void init() {
         super.init();
 
-        SshCommandSensor<String> hostnameSensor = new SshCommandSensor<String>(ConfigBag.newInstance()
-            .configure(SshCommandSensor.SENSOR_NAME, "fqdn")
-            .configure(SshCommandSensor.SENSOR_COMMAND, "hostname -s"
-        ));
-
         Object securityGroup = getConfig(SECURITY_GROUP);
-        setAttribute(AMBARI_SERVER, addChild(createServerSpec(hostnameSensor, securityGroup)));
-        
-        setAttribute(AMBARI_AGENTS, addChild(EntitySpec.create(DynamicCluster.class)
-            .configure(DynamicCluster.INITIAL_SIZE, getRequiredConfig(INITIAL_SIZE))
-            .configure(DynamicCluster.MEMBER_SPEC, createAgentSpec(hostnameSensor, securityGroup))
-            .displayName("All Nodes")
-        ));
-        
+        setAttribute(AMBARI_SERVER, addChild(createServerSpec(securityGroup)));
+
+        Integer initialSize = getRequiredConfig(INITIAL_SIZE);
+        if (initialSize > 0) {
```

Comment:
Make this clearer - we use initial size to decide whether we are using the recommendations api to define cluster or hostgroups from blueprint